lib/linux_admin/ssh_agent.rb

```diff
@@ -0,0 +1,94 @@
+require 'net/ssh'
+require 'fileutils'
+module LinuxAdmin
+  class SshAgent
+    attr_reader :pid
+    attr_reader :sock
+
+    def initialize(ssh_private_key, options = {})
+      @sock        = options[:agent_socket] if options[:agent_socket]
+      @private_key = ssh_private_key
+    end
+
+    def start
+      if @sock
+        FileUtils.mkdir_p(File.dirname(@sock))
+        agent_details = `ssh-agent -a #{@sock}`
+      else
+        agent_details = `ssh-agent`
+        @sock         = agent_details.split("=")[1].split(" ")[0].chop
+      end
+      @pid = agent_details.split("=")[2].split(" ")[0].chop
+      IO.popen({"SSH_AUTH_SOCK" => @sock, "SSH_AGENT_PID" => @pid}, ["ssh-add", "-"], :mode => 'w') do |f|
+        f.puts(@private_key)
+      end
+      if $?.to_i != 0
+        raise StandardError, "Couldn't add key to agent"
+      end
+    end
+
+    def stop
+      system({"SSH_AGENT_PID" => @pid}, "(ssh-agent -k) &> /dev/null") if process_exists?(@pid)
+      File.delete(@sock) if File.exist?(@sock)
+    end
+
+    def perform_commands(ip, username, commands = [])
+      result = nil
+      Net::SSH.start(ip, username, :paranoid => false, :forward_agent => true, :number_of_password_prompts => 0, :agent_socket_factory => -> { UNIXSocket.open(@sock) }) do |ssh|
+        if block_given?
+          result = yield ssh
+        else
+          commands.each do |cmd|
+            result = ssh_exec!(ssh, cmd)
+            result[:last_command] = cmd
+            break if result[:exit_status] != 0
+          end
+        end
+      end
+      result
+    end
+
+    private
+
+    def process_exists?(process_pid)
+      Process.kill(0, process_pid) == 1
```

Comment:
If that class existed I would say yes, but I don't think it's necessary to create it in this PR.